test_fakeredis.py

```diff
@@ -3337,6 +3337,22 @@ def test_eval_hgetall(self):
             [[b'k1', b'bar'], [b'k2', b'baz']]
         )
 
+    def test_eval_hgetall_iterate(self):
+        self.redis.hset('foo', 'k1', 'bar')
+        self.redis.hset('foo', 'k2', 'baz')
+        lua = """
+        local result = redis.call("hgetall", "foo")
+        for i, v in ipairs(result) do
+        end
+        return result
+        """
+        val = self.redis.eval(lua, 0, 'foo')
```

Comment:
Should be 1, not 0 (number of keys passed).

Reply:
You are totally right.
Fixed it.

Reply:
You are totally right.
Fixed it.